ops/testing.py

```diff
@@ -644,24 +703,27 @@ def remove_relation(self, relation_id: int) -> None:
         Raises:
             RelationNotFoundError: if relation id is not valid
         """
+        rel_names = self._backend._relation_names   # pyright:ReportPrivateUsage=false
         try:
-            relation_name = self._backend._relation_names[relation_id]
+            relation_name = rel_names[relation_id]
             remote_app = self._backend.relation_remote_app_name(relation_id)
         except KeyError as e:
             raise model.RelationNotFoundError from e
 
-        for unit_name in self._backend._relation_list_map[relation_id].copy():
+        rel_list_map = self._backend._relation_list_map  # pyright:ReportPrivateUsage=false
```

Comment:
I think that ignoring these warnings is the right thing to do for now.

But I think that they are representative of one of the problems: our backend assumes that the data it loads on instantiation never need to be modified. And that immutability would bite us, were we to ever write a version of Ops that persisted between hook runs.

At some point, we probably need to write some formal routines for accessing, modifying, and cleaning up this data. Done right, it would make our startup and spindown process a lot cleaner and more understandable.